joomla/joomla-cms
# Add new Terms view

### Summary of Changes
For some time now I have found it very helpful to have a Terms view in Smart Search, especially when trying to debug new Smart Search plugins.  I had a bit of time to spare during the holiday break so I polished it up and I'm presenting it here for possible inclusion as a new feature in Joomla 3.7.

As well as adding the Terms view, this PR also adds a Terms column to the Indexed Content view which shows the number of terms indexed for each content item and is also a link to the Terms view filtered by that content item.

This PR adds a terms_count column to the #__finder_links table to keep track of the number of terms indexed for each content item.  Because this causes an error in recent MySQL versions, I have also updated the default date for the datetime columns so the error isn't thrown.

I have added update files for Postgres and SQL Azure, but I have no idea if they are correct or not.

### Testing Instructions
* Apply this PR.
* Go to Extensions -> Manage -> Database and click Fix.
* Go to Components -> Smart Search.  Rebuild the index by clicking Clear Index, then Index.  Alternatively, use php cli/finder_indexer.php --purge to rebuild the index from the command line.
* Notice that the Indexed Content view now has an extra Terms column which shows the number of terms indexed for each content item.  The numbers are also clickable links that will take you to the Terms view.
* Take a look around the Terms view, clicking on stuff.  Hopefully, it will work as expected.

### Documentation Changes Required
If this PR is accepted, I'll add an appropriate help screen.  The PR adds a help key of JHELP_COMPONENTS_FINDER_MANAGE_TERMS which for en-GB will attempt to load the help screen at https://docs.joomla.org/Help37:Components_Finder_Manage_Terms
